fractal_server/app/runner/v2/_local_processes/executor.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 (C) Friedrich Miescher Institute for Biomedical Research and
+# University of Zurich
+#
+# Original authors:
+# Tommaso Comparin <tommaso.comparin@exact-lab.it>
+#
+# This file is part of Fractal and was originally developed by eXact lab S.r.l.
+# <exact-lab.it> under contract with Liberali Lab from the Friedrich Miescher
+# Institute for Biomedical Research and Pelkmans Lab from the University of
+# Zurich.
+"""
+Custom version of Python
+[ProcessPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor)).
+"""
+import os
+import signal
+import threading
+import time
+from concurrent.futures import ProcessPoolExecutor
+from concurrent.futures.process import BrokenProcessPool
+from typing import Callable
+from typing import Iterable
+from typing import Optional
+from typing import Sequence
+
+from ._local_config import get_default_local_backend_config
+from ._local_config import LocalBackendConfig
+
+
+class FractalProcessPoolExecutor(ProcessPoolExecutor):
+    def __init__(self, *args, shutdown_file=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.shutdown_file = shutdown_file
+        if shutdown_file:
+            self.shutdown_thread = threading.Thread(
+                target=self._check_file, daemon=True
+            )
+            self.shutdown_thread.start()
+
+    def _check_file(self):
+        while True:
+            if os.path.exists(self.shutdown_file):
+                self._terminate_processes()
+                os.remove(self.shutdown_file)  # remove shutdown file
```

Comment:
no need to remove the file (which serves as a proof that the shutdown was triggered)